pkg/trace/stats/concentrator.go

```diff
@@ -112,18 +115,26 @@ func (c *Concentrator) Stop() {
 	c.exitWG.Wait()
 }
 
-// EnvTrace contains input for the concentractor.
-type EnvTrace struct {
-	Trace WeightedTrace
-	Env   string
-}
-
 // Input specifies a set of traces originating from a certain payload.
 type Input struct {
-	Traces      []EnvTrace
+	Traces      []traceutil.ProcessedTrace
 	ContainerID string
 }
 
+// NewStatsInput allocates a stats input for an incoming trace payload
+func NewStatsInput(p *pb.TracerPayload, clientComputedStats bool, conf *config.AgentConfig) Input {
+	if clientComputedStats {
+		return Input{}
+	}
+	in := Input{Traces: make([]traceutil.ProcessedTrace, 0, len(p.Chunks))}
+	if !features.Has("disable_cid_stats") && conf.FargateOrchestrator != fargate.Unknown {
+		// only allow the ContainerID stats dimension if we're in a Fargate instance
+		// and it's not prohibited by the disable_cid_stats feature flag.
+		in.ContainerID = p.ContainerID
```

Comment:
unrelated to this PR, but we should also add support for container ID in tracer stats right?

Reply:
it's still under a feature flag but we should probably do it too